### PR TITLE
Fix: Util function to ensure list of dicts all have same keys for hek

### LINKED ISSUE #3238: hek client search method broken for SPoCA
### Description
Under sunpy version 1.0.1 installed from conda-forge, ```hek.HEKClient().search()``` issues ```KeyError``` and ```ValueError``` using ```hek.attrs.FRM.Name == 'SPoCA'```
I do not have this issue when using ```hek.attrs.FRM.Name == 'EGSO_SFC'``` for the same dates. 

### Expected behavior
No problem under sunpy 0.9.x
Also, I had no problem  when using one of the github dev versions at 1.0.x in the last week of May, I did not retain the version number. So this bug must have been introduced a few days ago before you officially released 1.0.1 and after the version RC4. That's all I can remember. 

### Steps to Reproduce

```
from sunpy.net import hek, helioviewer
from sunpy.time import parse_time
client = hek.HEKClient()
tstart = '2013/02/01 00:00:00'
tend = '2013/02/01 23:30:00'
time_start = parse_time(tstart)
time_end = parse_time(tend)
results = client.search(hek.attrs.Time(time_start, time_end), hek.attrs.FRM.Name == 'SPoCA')
```

### System Details

  - SunPy Version: 1.0.1 and dev version 1.1.x
  - Python Version: 3.6 & 3.7
  - OS information: macOS 10.13.6

### PR DESCRIPTION
**Pull request by nabobalis**
Fixes #3238 via a large HACK. 

Not really sure this is a sane way to do it however.

### CODE REVIEW

File: sunpy/util/util.py [hunk 1]
```diff
@@ -131,3 +132,35 @@ def partial_key_match(key, dictionary):
     for k, v in dictionary.items():
         if all(k1 == k2 or k2 is None for k1, k2 in zip(k, key)):
             yield v
+
+
+def dict_keys_same(list_of_dicts):
+    """
+    Makes sure that a list of dictinoaries all have the same keys.
```
Comment (Cadair) [inline sunpy/util/util.py]: typo

**Comment by Cadair** [comment]
So, let me check I understand this, some records that are coming back from the HEK have different (in table parlance) columns than others? So you are making sure all the rows contain the superset of all columns?

**Comment by nabobalis** [comment]
Yes.

**Comment by Cadair** [comment]
I think a different approach would be to use the `columns=` kwarg to Table, and just pass it a list of the superset of all columns. You could construct the set of all columns by doing set operations on the keys.

i.e.:
```python
>>> d1 = {1:2, 3:4}; d2 = {3:4, 5:6}                                                                                                                                                                                                               
>>> s = set()                                                                                                                                                                                                                                      
>>> s.union(*map(lambda x: x.keys(), [d1, d2]))                                                                                                                                                                                                    
{1, 3, 5}
```

I can not promise this will work though, but it would mean not having to actually change the list of dicts.

**Comment by nabobalis** [comment]
K

**Comment by Cadair** [comment]
Also the constants thing needs a changelog entry of its own.

File: changelog/3240.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Fixed HEKClient return when the results were a mixed set of dictonaries.
```
Comment (Cadair) [inline changelog/3240.bugfix.rst]: linkify `HEKClient`

Comment (nabobalis) [reply changelog/3240.bugfix.rst]: Done.

**Comment by nabobalis** [comment]
> Also the constants thing needs a changelog entry of its own.

Done.

**Comment by nabobalis** [comment]
Trying to fix that astropy dev failure.

**Comment by nabobalis** [comment]
What happens if you set the units and constants versions at the start of conftest.py

**Comment by nabobalis** [comment]
Database failures are back.

**Comment by nabobalis** [comment]
So I changed them to macos.

File: azure-templates.yml [hunk 3]
```diff
@@ -4,46 +4,11 @@ jobs:
     ${{ if eq(parameters.os, 'windows') }}:
       vmImage: vs2017-win2016
     ${{ if eq(parameters.os, 'macos') }}:
-      vmImage: macOS 10.13
+      vmImage: macOS 10.14
     ${{ if eq(parameters.os, 'linux') }}:
       vmImage: Ubuntu 16.04
 
   steps:
-  - ${{ if eq(parameters.os, 'linux') }}:
```
Comment (Cadair) [inline azure-templates.yml]: rejoice!

**Pull request merged**